---
title: "Class11"
author: "Sabrina Koldinger (A16368238)"
format: pdf
editor: visual
---
# Class11- Structural Bioinformatics pt.2

AlphaFold has changed the game for protein structure prediction allowing anyone with bioinfomatics skills to predict the structure of almost any protein:

We used : https://colab.research.google.com/github/sokrypton/ColabFold/blob/main/AlphaFold2.ipynb
Specificaly, we used the AlphaFold2_mmseqs2 version that uses mmseqs2 rater than HMMMer for sequence search.

We obtained PDB structure files and JSON format files to tell us the acuracy and quality of the structures.

Let's load the structures in Mol*. 

```{r}
library(bio3d)
results_dir <- "HIVPR2_23119/" 

```

```{r}
pdb_files <- list.files(path=results_dir,
                        pattern="*.pdb",
                        full.names = TRUE)

# Print our PDB file names
basename(pdb_files)
```

```{r}
pdbs <- pdbaln(pdb_files, fit=TRUE, exefile="msa")
```

```{r}
pdbs
```

## RMSD

```{r}
rd <- rmsd(pdbs, fit=T)
range(rd)
```

Heatmap
```{r}
library(pheatmap)

colnames(rd) <- paste0("m",1:5)
rownames(rd) <- paste0("m",1:5)
pheatmap(rd)
```



```{r}
pdb <- read.pdb("1hsg")
```
```{r}
plotb3(pdbs$b[1,], typ="l", lwd=2, sse=pdb)
points(pdbs$b[2,], typ="l", col="red")
points(pdbs$b[3,], typ="l", col="blue")
points(pdbs$b[4,], typ="l", col="darkgreen")
points(pdbs$b[5,], typ="l", col="orange")
abline(v=100, col="gray")
```

```{r}
core <- core.find(pdbs)
core.inds <- print(core, vol=0.5)
```


```{r}
xyz <- pdbfit(pdbs, core.inds, outpath="corefit_structures")
```

## RSMF
```{r}
rf <- rmsf(xyz)

plotb3(rf, sse=pdb)
abline(v=100, col="gray", ylab="RMSF")
```

## Predicted Alignment Error
```{r}
library(jsonlite)

# Listing of all PAE JSON files
pae_files <- list.files(path=results_dir,
                        pattern=".*model.*\\.json",
                        full.names = TRUE)
```

```{r}
pae1 <- read_json(pae_files[1],simplifyVector = TRUE)
pae5 <- read_json(pae_files[5],simplifyVector = TRUE)

attributes(pae1)
```

```{r}
head(pae1$plddt)
```


Plots of the first and fifth files.
```{r}
plot.dmat(pae1$pae, 
          xlab="Residue Position (i)",
          ylab="Residue Position (j)")

plot.dmat(pae5$pae, 
          xlab="Residue Position (i)",
          ylab="Residue Position (j)",
          grid.col = "black",
          zlim=c(0,30))
```

The first file looks far better than the fifth. 

```{r}
plot.dmat(pae1$pae, 
          xlab="Residue Position (i)",
          ylab="Residue Position (j)",
          grid.col = "black",
          zlim=c(0,30))
```


## Residue Conservation

```{r}
aln_file <- list.files(path=results_dir,
                       pattern=".a3m$",
                        full.names = TRUE)
aln_file
aln <- read.fasta(aln_file[1], to.upper = TRUE)
```

```{r}
dim(aln$ali)
sim <- conserv(aln)
```

```{r}
plotb3(sim[1:99], sse=trim.pdb(pdb, chain="A"),
       ylab="Conservation Score")
```

```{r}

```




